### ISSUE TITLE
[Bug] Broken emotes

### ISSUE THREAD

**Issue by Rmin-Sadigh** [SevenTV/SevenTV#252]
**Describe the bug**
Some emotes don't load and a broken image+emote name is shown instead.

**To Reproduce**
Look for similar behavior in twitch channels. It's a rare occurrence. Broken emotes are random per channel and most channels don't have any. Each channel's broken emotes always stay the same. Provided screenshots are from [Gorgc](https://www.twitch.tv/gorgc)'s channel where the LOLW and KEKW emotes have been broken for almost 8 months so far. These emotes load properly in other channels.
Right-clicking the broken image and clicking 'open image in a new tab' will open [this](https://www.twitch.tv/undefined) link every time for every broken emote.

**Screenshots**
![image](https://user-images.githubusercontent.com/61507499/164300208-7ce00341-1d9d-40b5-ba9c-1f636d08adf8.png)
The emote is correctly listed and viewed in the emote menu
![image](https://user-images.githubusercontent.com/61507499/164300765-df6e1378-3192-459b-a6f1-f0cb90d36f23.png)

**Version:**
Version: 2.2.1

**Comment by Rmin-Sadigh** [comment]
I tried 'inspect element' and this is the HTML code of a broken emote.
![image](https://user-images.githubusercontent.com/61507499/164911596-c6fb103e-e9ad-4ef6-894f-dd4db293abc9.png)

the 1x source URL is correct, but other sizes are set as undefined, which may be the reason clicking 'load image in new tab' redirects to [twitch.tv/undefined](twitch.tv/undefined). I tried loading the supposed URL of other sizes manually and they all have valid URLs and the emotes load in larger sizes.

**Comment by AnatoleAM** [comment]
This is because FrankerFaceZ does not actually provide responsive scales for these specific emotes. In the case of [PepeLaugh](https://www.frankerfacez.com/emoticon/64785-PepeLaugh) you can see that opening the "4x" link will actually take you to 1x.

This is unexpected behavior which the extension does not currently handle. This could be fixed by filtering out non-existing scales.

**Comment by Rmin-Sadigh** [comment]
> opening the "4x" link will actually take you to 1x.

The 4x image is loaded as a larger image, but yes the quality is so low that it seems like the 1x image has been resized.

> This could be fixed by filtering out non-existing scales.

Do I have to enable any settings, or this will be added later? I didn't find any options related to this.

**Comment by timmclean** [comment]
It's my first time looking at this codebase but I think I found the bug and the 1-line change needed to fix it.

Image URLs are retrieved by calling `Emote.cdn`. `Emote.cdn` actually already has handling for the case where a certain image resolution is missing on line 303:

https://github.com/SevenTV/SevenTV/blob/e6079c313654c69e978388e085675e9a6daa6244/src/Global/EmoteStore.tsx#L297-L317

It defaults to the last entry in the list, which I would guess is usually the highest available resolution.

That should address the situation in this bug, so why is `Emote.cdn` returning `undefined`? I think this is the answer:

https://github.com/SevenTV/SevenTV/blob/e6079c313654c69e978388e085675e9a6daa6244/src/Global/API.ts#L121-L143

The list of resolutions is hard-coded here, so entries are created even for missing resolutions: `["2", undefined]`, `["3", undefined]`, `["4", undefined]`. It looks like this could be fixed by filtering the `urls` list on line 136 with something like:

```js
    ].filter(([_, url]) => url), // filter out missing resolutions
```

Cheers!

**Comment by timmclean** [comment]
Not fixed yet for me as of 2.2.2

**Comment by Rmin-Sadigh** [comment]
> Has this been fixed for you with the latest updates?

I'm on v 2.2.2 and the problem still persists.

**Comment by timmclean** [comment]
Confirmed fixed for me as of the v2.2.3 release! Thanks :)
